csharp/src/iceboxnet/ServiceManagerI.cs

```diff
@@ -814,13 +711,19 @@ private enum ServiceStatus
             Started
         }
 
-        private struct ServiceInfo
+        private class ServiceInfo
         {
-            public string name;
-            public Service service;
-            public Ice.Communicator communicator;
-            public ServiceStatus status;
-            public string[] args;
+            public ServiceInfo(string name, ServiceStatus status, string[] args)
+            {
+                Name = name;
+                Status = status;
+                Args = args;
+            }
+            public string Name { get; private set; }
+            public ServiceStatus Status { get; set; }
+            public string[] Args;
```

Comment:
it would be nicer to not mix properties and fields

Reply:
Right, I can use a property, and copy it to be able to pass it to `ParseIceArgs`, not sure if we can overload `ParseIceArgs` to accept non `ref` args